backend/api/v2beta1/run.proto

```diff
@@ -436,4 +443,13 @@ message ReadArtifactRequest {
 message ReadArtifactResponse {
   // Byte array of the artifact content.
   bytes data = 1;
+}
+
+message RetryRunRequest {
+ // The ID of the parent experiment.
+ string experiment_id = 1;
```

Comment:
Remove `experiment_id`
Not presented in v1 either:

https://github.com/kubeflow/pipelines/blob/b3de27a7d01ee7febef58b92abc52ffb1f49b4a7/backend/api/v1beta1/run.proto#L180-L183

Reply:
The `experiment_id` is added to be consistent with other v2 methods in this proto, including [GetRunRequest](https://github.com/kubeflow/pipelines/blob/b3de27a7d01ee7febef58b92abc52ffb1f49b4a7/backend/api/v2beta1/run.proto#L345), [TerminateRunRequest](https://github.com/kubeflow/pipelines/blob/b3de27a7d01ee7febef58b92abc52ffb1f49b4a7/backend/api/v2beta1/run.proto#L379), [ArchiveRunRequest](https://github.com/kubeflow/pipelines/blob/b3de27a7d01ee7febef58b92abc52ffb1f49b4a7/backend/api/v2beta1/run.proto#L398), [UnarchiveRunRequest](https://github.com/kubeflow/pipelines/blob/b3de27a7d01ee7febef58b92abc52ffb1f49b4a7/backend/api/v2beta1/run.proto#L406), and [DeleteRunRequest](https://github.com/kubeflow/pipelines/blob/b3de27a7d01ee7febef58b92abc52ffb1f49b4a7/backend/api/v2beta1/run.proto#L414). All these messages include an `experiment_id` field because there is an `experiment_id` section in the API path ([example](https://github.com/kubeflow/pipelines/blob/b3de27a7d01ee7febef58b92abc52ffb1f49b4a7/backend/api/v2beta1/run.proto#L73)). I agree that here `experiment_id` is redundant, although it follows [AIP design principle](https://google.aip.dev/122). However, I do think we need to maintain consistency among all methods across the same service, i.e., either all methods within RunService should have `experiment_id` as part of the path, or no method does.

Reply:
Synced offline, given that experiment is a parent resource of run, we need to keep the experiment_id here. But the current implementation of ignoring experiment id should be fixed as well. This is not a blocker for the PR, consider the same pattern exists in GetRuns, GetJobs, etc.--although we do have validation that could complain "the run_id doesn't belong to the experiment" for those APIs, while such validation is missing for the retry run. 
We can fix the validation across all the APIs to check if "run id can be find in the given experiment id" post beta0 release.

Reply:
Before seeing your comment, I added logic to verify the experiment id is correct.